pynetbox/core/api.py

```diff
@@ -208,3 +210,31 @@ def create_token(self, username, password):
         # object details will fail
         self.token = resp["key"]
         return Record(resp, self, None)
+
+    @contextlib.contextmanager
+    def activate_branch(self, branch):
+        """
+        Context manager to activate the branch by setting the schema ID in the headers.
+
+        :Raises: ValueError if the branch is not a valid NetBox branch.
+
+        :Example:
+
+        >>> import pynetbox
+        >>> nb = pynetbox.api("https://netbox-server")
+        >>> branch = nb.plugins.branching.branches.create(name="testbranch")
+        >>> with nb.activate_branch(branch):
+        ...     sites = nb.dcim.sites.all()
+        ...     # All operations within this block will use the branch's schema
+        """
+        if not isinstance(branch, Record) or not "schema_id" in dict(branch):
```

Comment:
I was going to ask if this `isinstance()` check could be more explicit, but I don't think it can be without creating a model for `netbox_branching` objects in pynetbox, right?